src/crowdstrike/CrowdStrikeApiGateway.ts

```diff
@@ -375,17 +372,20 @@ export class CrowdStrikeApiGateway {
         paginationParams.offset = seen;
       }
 
-      const baseUrl = this.queryBuilder.buildResourcePathUrl(
-        availabilityZone,
-        resourcePath,
-      );
-
-      this.total.setValue(baseUrl, paginationParams?.total);
-
-      total = this.total.getValue(baseUrl);
+      if (paginationParams?.total) {
+        total = paginationParams.total;
+      }
 
       finished = seen === 0 || seen >= total;
 
+      if (!response.resources?.length && finished !== true) {
```

Comment:
nit: consider the change for readability
```suggestion
      if ((!response.resources || response.resources.length === 0) && !finished) {
```